firedrake/preconditioners/facet_split.py

```diff
@@ -143,11 +144,10 @@ def _restrict_nullspace(nsp):
 
     def update(self, pc):
         if hasattr(self, "_permute_op"):
```

Comment:
Please can we add a member for what configuration is being used instead of implicitly checking this based on what attributes we have? It would be clearer to the reader and more robust.